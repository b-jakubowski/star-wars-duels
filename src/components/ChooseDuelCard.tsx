import React from 'react';
import { makeStyles, Card, CardContent, CardHeader, CardMedia } from '@material-ui/core';
import { Link } from 'react-router-dom';

interface ChooseDuelCardProps {
	title: string;
	background: string;
	route: string;
}

const useStyles = makeStyles({
	link: {
		textDecoration: 'none',
	},
	root: {
		marginTop: 30,
	},
	media: {
		height: 250,
	},
	title: {
		textAlign: 'center',
	},
});

const ChooseDuelCard: React.FC<ChooseDuelCardProps> = ({ title, background, route }: ChooseDuelCardProps) => {
	const classes = useStyles();

	return (
		<Link className={classes.link} to={route}>
			<Card className={classes.root}>
				<CardContent>
					<CardHeader className={classes.title} title={title} />
					<CardMedia className={classes.media} image={background} />
				</CardContent>
			</Card>
		</Link>
	);
};

export default ChooseDuelCard;
