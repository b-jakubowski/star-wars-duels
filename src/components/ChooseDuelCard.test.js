import React from 'react';
import { HashRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import duelPic from '../images/duel-pic.jpg';
import starshipsPic from '../images/starships-pic.jpg';
import ChooseDuelCard from './ChooseDuelCard';

const componentHeroesProp = (
	<Router>
		<ChooseDuelCard route="/heroes" title="Hero vs. Hero" background={duelPic} />
	</Router>
);

const componentStarshipsProp = (
	<Router>
		<ChooseDuelCard route="/starships" title="Starship vs. Starship" background={starshipsPic} />
	</Router>
);

it('ChooseDuelCard renders correctly with heroes props', () => {
	const tree = renderer.create(componentHeroesProp).toJSON();
	expect(tree).toMatchSnapshot();
});

it('ChooseDuelCard renders correctly with starships props', () => {
	const tree = renderer.create(componentStarshipsProp).toJSON();
	expect(tree).toMatchSnapshot();
});
