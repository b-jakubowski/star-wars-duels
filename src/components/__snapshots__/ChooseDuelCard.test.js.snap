// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseDuelCard renders correctly with heroes props 1`] = `
<a
  className="makeStyles-link-1"
  href="#/heroes"
  onClick={[Function]}
>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-root-2 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardContent-root"
    >
      <div
        className="MuiCardHeader-root makeStyles-title-4"
        data-testid="card-title"
      >
        <div
          className="MuiCardHeader-content"
        >
          <span
            className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
          >
            Hero vs. Hero
          </span>
        </div>
      </div>
      <div
        className="MuiCardMedia-root makeStyles-media-3"
        data-testid="card-img"
        style={
          Object {
            "backgroundImage": "url(\\"duel-pic.jpg\\")",
          }
        }
      />
    </div>
  </div>
</a>
`;

exports[`ChooseDuelCard renders correctly with starships props 1`] = `
<a
  className="makeStyles-link-1"
  href="#/starships"
  onClick={[Function]}
>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-root-2 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardContent-root"
    >
      <div
        className="MuiCardHeader-root makeStyles-title-4"
        data-testid="card-title"
      >
        <div
          className="MuiCardHeader-content"
        >
          <span
            className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
          >
            Starship vs. Starship
          </span>
        </div>
      </div>
      <div
        className="MuiCardMedia-root makeStyles-media-3"
        data-testid="card-img"
        style={
          Object {
            "backgroundImage": "url(\\"starships-pic.jpg\\")",
          }
        }
      />
    </div>
  </div>
</a>
`;
